Lib/unittest/mock.py

```diff
@@ -402,6 +403,14 @@ def __init__(self, /, *args, **kwargs):
 class NonCallableMock(Base):
     """A non-callable version of `Mock`"""
 
+    # Store a mutex as a class attribute in order to protect concurrent access
+    # to mock attributes. Using a class attribute allows all NonCallableMock
+    # instances to share the mutex for simplicity.
+    #
+    # See https://github.com/python/cpython/issues/98624 for why this is
+    # necessary.
+    _lock = RLock()
```

Comment:
@ambv could that be renamed? this change broke us since we no longer get an implicit our_object._lock mock... Can be worked around, but potentially affect a lot of people.